### PR TITLE
#9852 Events participants created as archived

### PR DESCRIPTION
**Pull request by andrei7bal**
#9852 Event participants newly created for archived events are archived too, addressing feedback of #10413

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java [hunk 1]
```diff
@@ -357,6 +357,7 @@ public EventParticipantDto saveEventParticipant(@Valid EventParticipantDto dto,
 			// The Event Participant is newly created, let's check if the related person is related to other events
 			// In that case, let's notify corresponding responsible Users of this relation
 			notifyEventResponsibleUsersOfCommonEventParticipant(entity, event);
+			entity.setArchived(event.isArchived());
```
Comment (StefanKock) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java:360]: The change should be done before `service.ensurePersisted`, maybe within `fillOrBuildEntity` or directly afterwards.

File: sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbTest.java [hunk 2]
```diff
@@ -551,6 +552,38 @@ public void searchEventParticipantsByPersonOtherDetail() {
 		eventParticipantIndexDtos = getEventParticipantFacade().getIndexList(eventParticipantCriteria, 0, 100, null);
 		assertEquals(0, eventParticipantIndexDtos.size());
 	}
+	@Test
+	public void testCreationOfEventParticipantWhenArchived()
+	{
```
Comment (StefanKock) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbTest.java:557]: Code formatting violated: Use Code formatting to move `{` after the method declaration

File: sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbTest.java [hunk 3]
```diff
@@ -551,6 +552,38 @@ public void searchEventParticipantsByPersonOtherDetail() {
...
+		assertTrue(getEventParticipantFacade().isArchived(archivedEventParticipantDto.getUuid()));
+
+		getEventFacade().dearchive(Collections.singletonList(event.getUuid()), "reason");
+		PersonDto eventPersonActive = creator.createPerson("EventActive", "OrganizerActive");
+		EventParticipantDto activeEventParticipantDto = creator.createEventParticipant(event.toReference(), eventPersonActive, "event Director", user.toReference());
+		assertFalse(getEventParticipantFacade().isArchived(activeEventParticipantDto.getUuid()));
+	}
+
+	@Test
+	public void testCreationOfExistingEventParticipantRemainsArchived()
```
Comment (StefanKock) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbTest.java:573]: This test is almost identical with the first test case. Please join both test cases (saves several seconds for unit test setup), and add code comments before each block to explain what that test step is testing. Example:

```java
// Verify that dearchiving of the event does not dearchive the event participant
```

File: sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbTest.java [hunk 4]
```diff
@@ -551,6 +552,38 @@ public void searchEventParticipantsByPersonOtherDetail() {
 		eventParticipantIndexDtos = getEventParticipantFacade().getIndexList(eventParticipantCriteria, 0, 100, null);
 		assertEquals(0, eventParticipantIndexDtos.size());
 	}
+	@Test
+	public void testCreationOfEventParticipantWhenArchived()
```
Comment (StefanKock) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbTest.java:556]: Please rename unit test to `testSaveEventParticipantWithEventArchiving`

**Review by StefanKock — CHANGES_REQUESTED** [review]

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
The change done in this PR is not acceptable. We can't set the archived status of event participants to the archived status of events all the time. There's the very likely scenario of an active event with some active and some archived event participants; editing one of the archived event participants retrospectively must NOT change their status back to active.

As discussed before, please ensure that:
- There's code in the logic that archives/de-archives an event that archives/de-archives all its event participants in the same process.
- Make sure that when saving an event participant for an ARCHIVED event, it's always archived.

**Review by MateStrysewske — APPROVED** [review]

File: sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbTest.java [hunk 5]
```diff
@@ -551,6 +552,28 @@ public void searchEventParticipantsByPersonOtherDetail() {
...
+		EventParticipantDto archivedEventParticipantDto = creator.createEventParticipant(event.toReference(), eventPersonArchived, "event Director", user.toReference());
+		assertTrue(getEventParticipantFacade().isArchived(archivedEventParticipantDto.getUuid()));
+
+		// Validate that for a newly created Event Participant the active status is kept from the Event
+		getEventFacade().dearchive(Collections.singletonList(event.getUuid()), "reason");
+		PersonDto eventPersonActive = creator.createPerson("EventActive", "OrganizerActive");
+		EventParticipantDto activeEventParticipantDto = creator.createEventParticipant(event.toReference(), eventPersonActive, "event Director", user.toReference());
+		assertFalse(getEventParticipantFacade().isArchived(activeEventParticipantDto.getUuid()));
+
+		//Validate that for an already existing Event Participant the previous status is kept
```
Comment (StefanKock) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbTest.java:572]: This test description does not match: Creating a new event participant does not relate to what happened to the `archivedEventParticipantDto`

The last `assertEquals(getEventParticipantFacade().isArchived(archivedEventParticipantDto.getUuid()), getEventParticipantFacade().isArchived(alreadyArchivedParticipantDto.getUuid()));` does not check the expected behaviour but only if two values in the database are the same (but not if true or false).

**Review by StefanKock — CHANGES_REQUESTED** [review]
Only a finding on a curious test, the relevant changes look good

**Review by StefanKock — APPROVED** [review]

**Pull request merged**